api/proto/config/storage.proto

```diff
@@ -106,3 +106,54 @@ message Drive {
   // A value of 0 will indicate that no resizing is required
   int64 maxsizebytes = 10;
 }
+
+
+message ContentTree {
+  string uuid = 1;
+  string dsId  = 2; // effectively pointer/key into dsConfigs
+
+  string URL = 3;
+  Format iformat = 4;
+
+  // the following is only used for individual blobs, if this message references a group of blobs
+  // e.g. in the case of OCI -- this information is expected to be provided by a top level blob
+  // that this message points to via its URL
+  string sha256 = 5;
+  uint64 maxSizeBytes = 6; // used for capping resource consumption eve for OCI & BLOB_TREE
+  SignatureInfo siginfo = 7;
+}
+
+enum VolumeAccessProtocols {
+  VAP_NONE = 0;
+  VAP_9P = 1;
+}
+
+enum VolumeContentOriginType {
+  VCOT_UNKNOWN = 0;
+  VCOT_BLANK = 1;
+  VCOT_DOWNLOAD = 2;
+}
+
+message VolumeContentOrigin {
+  volumeContentOriginType type = 1;
+  repeated string downloadContentTreeID = 2;
```

Comment:
I don't think this should be repeated. Does it say so in the document?

Would make sense to add a comment for the field saying that it applies when the type is VOT_DOWNLOAD

Reply:
Furthermore, since the contentTreeID does not identify a particular sha, I think we should add a downloadContentTreeSha in here.

Reply:
I removed the repeated and added the comment.

Regarding adding a sha, there is already sha256 in message ContentTree. If that doesn't suffice (for e.g., cases like ":latest"), the controller does not know the sha256 and can't give it in the config. I can add sha256 to message ZInfoContentTree. If you are thinking of some other case, let me know which message you think should have downloadContentTreeSha

Reply:
Ah - right. Doesn't make sense in the config. But does make sense in the info message.